frontend/src/TransitPredictionsPage.js

```diff
@@ -112,29 +112,50 @@ export default function TransitPredictionsPage() {
-                {stopName}
+            {stopOptions.map((stop) => (
+              <option key={stop.name} value={stop.name}>
+                {`${stop.name} (${stop.ids.length > 0 ? stop.ids.join(', ') : 'No IDs'})`}
               </option>
             ))}
           </select>
         </div>
         {/* Tabs for larger screens */}
         <div className="hidden items-center xl:flex flex-row justify-center space-x-4 overflow-x-auto">
-          {stopOptions.map((stopName) => (
-            <button key={stopName} className={`px-4 py-2 text-xl font-semibold border-b-2 
-            ${selectedStop === stopName ? 'border-blue-500 text-blue-500' : 'border-transparent text-gray-500 hover:border-gray-300'}`}
-              onClick={() => changeTab(stopName)}>
-              {stopName}
+          {stopOptions.map((stop) => (
+            <button key={stop.name} className={`px-4 py-2 text-xl font-semibold border-b-2 
+            ${selectedStop === stop.name ? 'border-blue-500 text-blue-500' : 'border-transparent text-gray-500 hover:border-gray-300'}`}
+              onClick={() => changeTab(stop.name)}>
+              {`${stop.name} `}
             </button>
           ))}
         </div>
+
         <div className="flex flex-col xl:grid xl:grid-cols-[70%_30%] gap-8">
           {!!busData.length && busData.map((stop) => (
             stop.name === selectedStop &&
-            <div key={stop.name} className="items-center text-center md:text-left flex flex-col w-full xl:w-full mt-2 p-6 min-w-50 max-h-[65vh] text-xl
-                bg-white rounded-lg border border-gray-200 shadow-md dark:bg-gray-800 dark:border-gray-700 overflow-y-auto">
-              <div className="flex justify-center xl:justify-start font-bold text-[1rem] md:text-4xl mb-5 xl:mb-10">{stop.name}</div>
+            <div key={stop.name} className="items-center text-center md:text-left flex flex-col w-full xl:w-full mt-2 p-6 min-w-50 max-h-[65vh] text-xl bg-white rounded-lg border border-gray-200 shadow-md dark:bg-gray-800 dark:border-gray-700 overflow-y-auto">
+              
+              <div className="flex justify-center xl:justify-start font-bold text-[1rem] md:text-4xl mb-2 xl:mb-5">
+                {stop.name}
+              </div>
+              <div className="flex justify-center xl:justify-start text-sm md:text-lg text-gray-500 mb-4 xl:mb-6">
+                {stop.ids.length > 0 ? (
+                  <>
+                    {stop.ids.some(id => !isNaN(id)) ? (
+                      <a
+                        href={`tel:511p1p1,,${stop.ids.find(id => !isNaN(id))}`}
+                        className="inline-block px-2 py-1 ml-2 text-white bg-blue-600 border border-blue-600 rounded hover:bg-blue-700"
+                      >
+                        CLICK TO CALL
+                      </a>
+                    ) : (
+                      <span className="inline-block px-2 py-1 ml-2 text-gray-500 border border-gray-500 rounded">NO CALL AVAILABLE</span>
```

Comment:
instead lets just not show anything if the call isnt available

Reply:
fixed